test/component/newstorage/newstore.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package newstorage contains common tests for newstorage implementation.
+//
+package newstorage
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/hyperledger/aries-framework-go/pkg/newstorage"
+	"github.com/stretchr/testify/require"
+)
+
+// TestAll tests common storage functionality.
+func TestAll(t *testing.T, provider newstorage.Provider) {
```

Comment:
When I work on other storage implementations in this repo, I'll add more common tests here